### PR TITLE
Revert "Horrible hack to work around bug in OCCT7.7.2"

### PR DESCRIPTION
**Pull request by yorikvanhavre**
This reverts commit 60640fa4415548ebfa19c1f929ea2da7e4980fe7 from PR #15931

### CODE REVIEW

**Comment by maxwxyz** [comment]
https://github.com/FreeCAD/FreeCAD/issues/15599 should be reopened then?

**Comment by kadet1090** [comment]
Is there a reason for the revert?

**Comment by maxwxyz** [comment]
@kadet1090 maybe https://forum.freecad.org/viewtopic.php?p=787298#p787298

**Comment by oursland** [comment]
@kadet1090 wwmayer identified that it solved one problem and created another: https://github.com/FreeCAD/FreeCAD/pull/15931#issuecomment-2416966752

**Comment by wwmayer** [comment]
> https://github.com/FreeCAD/FreeCAD/issues/15599 should be reopened then?

No. To fix the problem some user interaction is required. He has to use the transform tool for the cylinder and rotate by some degrees (ideally by 90 deg to get a nice looking model) around the Z axis. This makes sure to avoid co-planar faces between the two objects and helps the boolean intersection algorithm to work correctly.

**Comment by maxwxyz** [comment]
to be merged after CI

File: src/Mod/Part/App/PrimitiveFeature.cpp [hunk 1]
```diff
@@ -665,7 +665,7 @@ App::DocumentObjectExecReturn *Cone::execute()
             BRepPrimAPI_MakeCone mkCone(Radius1.getValue(),
                                         Radius2.getValue(),
                                         Height.getValue(),
-                                        Angle.getValue()/180.0f*M_PI);
+                                        Base::toRadians<double>(angle));
```
Comment (hyarion) [inline src/Mod/Part/App/PrimitiveFeature.cpp:668]: @yorikvanhavre CI is failing due to an error in solving a merge conflict here.
This line should be:
```c++
                                        Base::toRadians<double>(Angle.getValue()));
```

**Review by hyarion — CHANGES_REQUESTED** [review]

File: src/Mod/Part/App/PrimitiveFeature.cpp (hunk 1 above)
Comment (yorikvanhavre) [reply src/Mod/Part/App/PrimitiveFeature.cpp]: Thanks! Indeed that looked wrong :sweat_smile:

**Pull request merged**